FlagsmithClient/src/main/java/com/flagsmith/internal/FlagsmithRetrofitService.kt

```diff
@@ -49,31 +52,65 @@ interface FlagsmithRetrofitService {
                 }
             }
 
-            fun envKeyInterceptor(environmentKey: String): Interceptor {
+            fun jsonContentTypeInterceptor(): Interceptor {
                 return Interceptor { chain ->
-                    val request = chain.request().newBuilder()
-                        .addHeader("X-environment-key", environmentKey)
-                        .build()
-                    chain.proceed(request)
+                    val request = chain.request()
+                    if (chain.request().method == "POST" || chain.request().method == "PUT" || chain.request().method == "PATCH") {
+                        val newRequest = request.newBuilder()
+                            .header("Content-Type", "application/json; charset=utf-8")
+                            .header("Accept", "application/json")
+                            .build()
+                        chain.proceed(newRequest)
+                    } else {
+                        chain.proceed(request)
+                    }
+                }
+            }
+
+            fun updatedAtInterceptor(tracker: FlagsmithEventTimeTracker): Interceptor {
+                return Interceptor { chain ->
+                    val response = chain.proceed(chain.request())
+                    val updatedAtString = response.header("x-flagsmith-document-updated-at")
+                    Log.i("Flagsmith", "updatedAt: $updatedAtString")
+
+                    // Update in the tracker (Flagsmith class) if we got a new value
+                    tracker.lastFlagFetchTime = updatedAtString?.toDoubleOrNull() ?: tracker.lastFlagFetchTime
+                    return@Interceptor response
                 }
             }
 
+            val cache = if (context != null && cacheConfig.enableCache) Cache(context.cacheDir, cacheConfig.cacheSize) else null
+
             val client = OkHttpClient.Builder()
                 .addInterceptor(envKeyInterceptor(environmentKey))
+                .addInterceptor(updatedAtInterceptor(timeTracker))
+                .addInterceptor(jsonContentTypeInterceptor())
                 .let { if (cacheConfig.enableCache) it.addNetworkInterceptor(cacheControlInterceptor()) else it }
                 .callTimeout(requestTimeoutSeconds, java.util.concurrent.TimeUnit.SECONDS)
                 .readTimeout(readTimeoutSeconds, java.util.concurrent.TimeUnit.SECONDS)
                 .writeTimeout(writeTimeoutSeconds, java.util.concurrent.TimeUnit.SECONDS)
-                .cache(if (context != null && cacheConfig.enableCache) okhttp3.Cache(context.cacheDir, cacheConfig.cacheSize) else null)
+                .cache(cache)
                 .build()
 
+            // Make sure that we start with a clean cache
+            client.cache?.evictAll()
```

Comment:
I'm not sure which scenarios we would end up without a clean cache here since we're constructing it above?

Reply:
Hi @matthewelwell the cache is on-disk, so would survive a restart, hence the clear

Reply:
Understood.